### ISSUE TITLE
`parallel` function for running `cmd`s in parallel

### ISSUE THREAD

**Issue by SheepTester** [nbuilding/N-lang#188]
I think it'd be nice to be able to take advantage of the asynchronousness of `cmd`s by being able to execute another `cmd` in parallel.

Perhaps we can add a function `parallel`:

```ts
assert type parallel : [t] cmd[t] -> cmd[cmd[t]]

import time
let func1 = [] -> cmd[()] {
  time.sleep(1)!
  print("One second later")
}
let func2 = [] -> cmd[()] {
  time.sleep(2)!
  print("Two seconds later")
}
let pub main = [] -> cmd[()] {
  let func1Done = parallel(func1())!
  let func2Done = parallel(func2())!
  
  func1Done!
  func2Done!
}
  |> ()
```

This will wait a second, print `One second later`, wait another second (rather than two seconds), then print `Two seconds later`.

You give `parallel` a cmd, and once you await it with `!`, it'll start executing the given command in parallel. The cmd it returns, when awaited, will resolve once the given cmd finishes executing. If you await it after the cmd already finished, it'll resolve immediately.

Of course, there is no obligation to use the return value of `parallel`; you can let it run in parallel and not bother about it.

```ts
parallel(otherCmd)!
```

This can be used to implement setInterval (by using while true + time.sleep). Also, the nice thing about N is that most things are pure, so this can be used for multithreading, though currently communication between parallel cmds is limited. That'll have to go in a separate PR (what if we added [Mutex](https://doc.rust-lang.org/std/sync/struct.Mutex.html) to N :eyes:)

**Comment by Ashvin-Ranjan** [comment]
Closed via #203
